### ISSUE TITLE
AttributeError: 'Agent' object has no attribute 'AI_PROVIDER'

### ISSUE THREAD

**Issue by kanslor** [Josh-XT/Agent-LLM#306]
### Description

When starting a new task:

=== backend container Console output:
INFO:     172.18.0.1:50594 - "POST /api/chain/hello/run HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/uvicorn/protocols/http/httptools_impl.py", line 435, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/usr/local/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/fastapi/applications.py", line 276, in __call__
    await super().__call__(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/usr/local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.10/site-packages/starlette/middleware/cors.py", line 92, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "/usr/local/lib/python3.10/site-packages/starlette/middleware/cors.py", line 147, in simple_response
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/usr/local/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/usr/local/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/usr/local/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/usr/local/lib/python3.10/site-packages/fastapi/routing.py", line 237, in app
    raw_response = await run_endpoint_function(
  File "/usr/local/lib/python3.10/site-packages/fastapi/routing.py", line 163, in run_endpoint_function
    return await dependant.call(**values)
  File "/app/app.py", line 296, in run_chain
    Chain().run_chain(chain_name)
  File "/app/Chain.py", line 194, in run_chain
    AgentLLM(agent_name).run_chain_step(step_data)
  File "/app/AgentLLM.py", line 26, in __init__
    self.memories = Memories(self.agent_name, self.CFG)
  File "/app/Memories.py", line 20, in __init__
    if self.CFG.AI_PROVIDER == "openai":
AttributeError: 'Agent' object has no attribute 'AI_PROVIDER'

### Steps to Reproduce the Bug

git pull
docker-compose up -build
navigate to localhost:3000 in web browser
add new agent, set provider to oobabooga
set agent uri, context length, temp, model
test instruction，model works fine
enter task for agent "write me a short story"
click start pursuing task
nothing happens

### Expected Behavior

To start trying to complete the task the agent is given with the local install of oogabooga text generation.

### Actual Behavior

If you start docker without -d you can see the backend fail with the above error.

### Operating System

- [ ] Microsoft Windows
- [ ] Apple MacOS
- [X] Linux
- [ ] Android
- [ ] iOS
- [ ] Other

### Python Version

- [ ] Python <= 3.9
- [X] Python 3.10
- [ ] Python 3.11

### Environment Type - Connection

- [X] Local
- [ ] Remote

### Environment Type - Container

- [X] Using Docker
- [ ] Not Using Docker

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of Agent-LLM.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by Josh-XT** [comment]
Can you create a new agent to test with?  Maybe use something like huggingchat as the provider just to see.

**Comment by localagi** [comment]
Currently reproducing the issue
- using latest https://github.com/oobabooga/text-generation-webui
- using `--api` and `--listen` (for access from docker container
- creating agent on `:5000` block API, leaving everything else blank (defaults)

...now it's calculating

**Comment by localagi** [comment]
![image](https://github.com/Josh-XT/Agent-LLM/assets/132956819/b5de3aaf-b194-46fc-9a76-29fe71923c41)

At least the connection to the oobabooga works.

**Comment by localagi** [comment]
Adding `--verbose` to `text-generation-webui` allows to retrieve the request from `Agent-LLM`

```
"POST /api/v1/generate HTTP/1.1" 200 -


You are an AI who performs one task based on the following objective: write me a short story

You have the following constraints:
1. ~500 word limit for short term memory. Your short term memory is short, so immediately save important information to files.
2. If you are unsure how you previously did something or want to recall past events, thinking about similar events will help you remember.
3. No user assistance.

Take into account these previously completed tasks: 
Your task: Develop a task list to complete the objective if necessary.  The plan is 'None' if not necessary.

You have the following commands available to complete this task given.
Google Search - google_search({'query': None, 'num_results': 8})
Google Official Search - google_official_search({'query': None, 'num_results': 8})
Execute Python File - execute_python_file({'file': None})
Scrape Text with Playwright - scrape_text({'url': None})
Scrape Links with Playwright - scrape_links({'url': None})
Read Audio from File - read_audio_from_file({'audio_path': None})
Read Audio - read_audio({'audio': None})
Is Valid URL - is_valid_url({'url': None})
Sanitize URL - sanitize_url({'url': None})
Get Response - get_response({'url': None, 'timeout': 10})
Scrape Text - scrape_text({'url': None})
Scrape Links - scrape_links({'url': None})
Create Message - create_message({'chunk': None, 'question': None})
Check Duplicate Operation - check_duplicate_operation({'operation': None, 'filename': None})
Read File - read_file({'filename': None})
Ingest File - ingest_file({'filename': None, 'memory': None, 'max_length': 4000, 'overlap': 200})
Search Files - search_files({'directory': None})
Send Tweet - send_tweet({})
Browse Website - scrape_text_with_selenium({'url': None})
Get Datetime - get_datetime({})
Searx Search - search_searx({'query': None, 'category': 'general'})
Speak with TTS - speak({'text': None, 'engine': 'gtts', 'voice_index': 0})
Evaluate Code - evaluate_code({'code': None, 'agent_name': 'Agent-LLM'})
Analyze Pull Request - analyze_pull_request({'pr_url': None, 'agent_name': 'Agent-LLM'})
Perform Automated Testing - perform_automated_testing({'test_url': None, 'agent_name': 'Agent-LLM'})
Improve Code - improve_code({'suggestions': None, 'code': None, 'agent_name': 'Agent-LLM'})
Write Tests - write_tests({'code': None, 'focus': None, 'agent_name': 'Agent-LLM'})
Generate Image - generate_image({'prompt': None})
Create a new command - create_command({'function_description': None, 'agent_name': 'Agent-LLM'})
Ask AI Agent ooobooo - ask({'prompt': None, 'agent_name': 'Agent-LLM'})
Instruct AI Agent ooobooo - instruct({'prompt': None, 'agent_name': 'Agent-LLM'})

RESPOND IN THE FOLLOWING JSON FORMAT ONLY! THERE ARE SERIOUS CONSEQUENCES FOR BREAKING THIS RULE!

{
    "plan": "Your plan for achieving the task",
    "response": "Your response to the task",
    "commands": {
        "command_name": {
            "arg1": "val1",
            "arg2": "val2"
        },
        "command_name2": {
            "arg1": "val1",
            "arg2": "val2",
            "argN": "valN"
        }
    }
}

--------------------
```

**Comment by localagi** [comment]
@kanslor Check your chain at `/api/chain/hello/run` - it probably tries to enable another agent, not oobabooga.

**Comment by kanslor** [comment]
I set the chain by frontend, Maybe the chain lose the agent configuration. I'll try without docker environment to check. @localagi

**Comment by localagi** [comment]
Updated docs for oobabooga also in https://github.com/Josh-XT/Agent-LLM/pull/317

**Comment by kanslor** [comment]
sorry， I got a mistake between task and chain。I add a new chain， instead of task by frontend , when I run this chain，I found  that the console shows agent name(AgentLLM) isn't the name chain settings(oobabooga)，Maybe the chain setting have the problem cause this error.

**Comment by mongolu** [comment]
I have the same error with `v1.1.29-beta` but it works in `v1.1.28-beta`

Also in browser console it throws the error:
`Access to XMLHttpRequest at 'http://localhost:7437/api/agent' from origin 'http://localhost:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.`

**Comment by localagi** [comment]
@mongolu please open a new issue and append all information in detail on it. Describe your own setup there more extensively so we can reproduce.
@Josh-XT please close, i think its a false negative here.

**Comment by Josh-XT** [comment]
https://github.com/Josh-XT/Agent-LLM/commit/abefe132f5e53569faa80b6071b9771ced3edd74
See last commit and let us know if still having issues.
